### PR TITLE
fix(@cypress/react): Devtools unpredictable resets

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
This PR reduces unpredictable resets of the dev tools plugin. The main problem for our UI plugin architecture right now is our react components structure which sometimes recreates the whole subtree (of react-split-pane) and the root element of devtools gets recreated and replaced with a completely new one – so we see it empty. 

> But somewhere in memory there is still a pointer to the dom element that gets updated via devtools 🤦‍♂️

Devtools will flush also if the bridge is stale, but this happens only if we are not touching nor dev tools nor react app for some period of time, so I don't think this a problem.

### CODE REVIEW

**Comment by lmiller1990** [comment]
I confirmed you can open/close the devtools many times without problem.

I see we will have the bug where it renders a `<RunnerCt>` for each test. Maybe unrelated to this PR.

image

**Review by lmiller1990 — DISMISSED** [review]
The fix seems fine, I found another bug (perhaps existing, perhaps regression) and added a comment.

I'll leave fixing and/or merging up to @dmtrKovalenko. For now ✔️ for the fix.

**Comment by dmtrKovalenko** [comment]
Thanks 🙏 for finding a regression.

**Review by lmiller1990 — COMMENTED** [review]
Found a regression/potential issue, left a comment.

**Comment by dmtrKovalenko** [comment]
This issue reproducing again in `develop`
image

**Comment by dmtrKovalenko** [comment]
Hey @elevatebart I removed a bunch of code from your PR #15275 because it was a reason of duplicated components in devtools

File: packages/runner-ct/static/index.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE html>
 <html>
   <head>
+      <!-- Uncomment to connect using standalone react-devtools -->
+      <!-- <script src="http://localhost:8097"></script> -->
```
Comment (elevatebart) [inline packages/runner-ct/static/index.html:5]: Should we have this documented somewhere ?

File: npm/webpack-dev-server/src/aut-runner.ts [hunk 2]
```diff
@@ -21,22 +21,11 @@ export function init (importPromises, parent = (window.opener || window.parent))
   Cypress.onSpecWindow(window, importPromises)
   Cypress.action('app:window:before:load', window)
```
Comment (elevatebart) [inline npm/webpack-dev-server/src/aut-runner.ts:23]: The header is not what is hurting the dev tools

```suggestion
  // In this variable, we save head
  // innerHTML to account for loader installed styles
  let headInnerHTML = ''

  // before the run starts save
  Cypress.on('run:start', () => {
    headInnerHTML = document.head.innerHTML
  })
```

File: npm/webpack-dev-server/src/aut-runner.ts [hunk 3]
```diff
@@ -21,22 +21,11 @@ export function init (importPromises, parent = (window.opener || window.parent))
...
-  })
-
   // Before all tests we are mounting the root element, **not beforeEach**
   // Cleaning up platform between tests is the responsibility of the specific adapter
   // because unmounting react/vue component should be done using specific framework API
   // (for devtools and to get rid of global event listeners from previous tests.)
   Cypress.on('test:before:run', () => {
-    document.body.innerHTML = ''
-    document.head.innerHTML = headInnerHTML
     appendTargetIfNotExists('__cy_root')
```
Comment (elevatebart) [inline npm/webpack-dev-server/src/aut-runner.ts:29]:
```suggestion
    document.head.innerHTML = headInnerHTML
    appendTargetIfNotExists('__cy_root')
```

**Comment by lmiller1990** [comment]
@dmtrKovalenko Seems that removing some clean up breaks a spec in Vue where it wants to reset the state each test. I am not sure how important this test actually is, though.

Can give it another test when you are happy with it, pls let me know.

**Review by elevatebart — APPROVED** [review]

File: npm/webpack-dev-server/src/aut-runner.ts [hunk 4]
```diff
@@ -21,11 +29,22 @@ export function init (importPromises, parent = (window.opener || window.parent))
...
+  Cypress.on('run:start', () => {
+    headInnerHTML = document.head.innerHTML
+  })
+
   // Before all tests we are mounting the root element, **not beforeEach**
   // Cleaning up platform between tests is the responsibility of the specific adapter
   // because unmounting react/vue component should be done using specific framework API
   // (for devtools and to get rid of global event listeners from previous tests.)
   Cypress.on('test:before:run', () => {
+    document.head.innerHTML = headInnerHTML
```
Comment (dmtrKovalenko) [inline npm/webpack-dev-server/src/aut-runner.ts]: Sorry, @elevatebart  I appreciate your help, but I still don't understand why this should be a part of webpack-dev-server. It looks like not the job of the dev server. 

If somebody is going to not use our framework – let him cleanup his code by his own. It just looks like trying to solve a problem that is not exists.

For reference: testing library provides auto cleanup just like we do, but if you do can also use it manually I think for tests like this 

https://testing-library.com/docs/svelte-testing-library/api/#cleanup

**Review by lmiller1990 — APPROVED** [review]
This is working way better now. Great!

**Pull request merged**